src/traversals/greedy_color.jl

```diff
@@ -0,0 +1,155 @@
+"""
+    struct coloring{T}
+
+Store number of colors used and mapping from vertex to color
+"""
+struct coloring{T<:Integer}
+    num_colors::T
+    colors::Vector{T}
+end
+
+best_color(c1::coloring, c2::coloring) = c1.num_colors < c2.num_colors ? c1 : c2
+
+"""
+    perm_greedy_color(g, seq)
+
+Color graph `g` according to an order specified by `seq` using a greedy heuristic.
+seq[i] = v imples that vertex v is the i<sup>th</sup> vertex to be colored.
+"""
+function perm_greedy_color(
+    g::AbstractGraph,
+    seq::Vector{T}
+    ) where T <: Integer 
+
+    nvg::T = nv(g)
+    cols = Vector{T}(nvg)  
+    seen = zeros(Bool, nvg + 1)
+
+    for v in seq
+        seen[v] = true
+        colors_used = zeros(Bool, nvg)
+
+        for w in neighbors(g, v)
+            if seen[w]
+                colors_used[Int(cols[w])] = true
+            end
+        end
+
+     
+        for i in one(T):nvg
+            if colors_used[i] == false
+                cols[v] = i
+                break;
+            end
+        end
+    end
+
+    result = coloring{T}(maximum(cols), cols)
+    return result
+end
+
+"""
+    degree_greedy_color(g)
+
+Color graph `g` iteratively in the descending order of the degree of the vertices.
+"""
+function degree_greedy_color(g::AbstractGraph{U}) where U<:Integer
+    seq::Vector{U} = sortperm(degree(g, vertices(g)) , rev=true)
+    result = perm_greedy_color(g, seq)
+    return result
+end
+
+"""
+    parallel_random_greedy_color(g, reps)
+
+Color graph `g` iteratively in a random order using a greedy heuristic and
+choose the best coloring out of `reps` number of colorings computed in parallel.
+"""
+function parallel_random_greedy_color(
+    g::AbstractGraph,
+    reps::Integer
+    ) 
+
+    best::coloring = @parallel (best_color) for i in 1:reps
+        seq = shuffle(vertices(g))
```

Comment:
This may not be an issue, but check threads / forums for cautions regarding randomness in parallel.

Reply:
In  https://github.com/JuliaLang/julia/issues/94 it states that each processor starts with different seeds. Will that be sufficient for this problem?

Reply:
If you can confirm we get different seeds, then sure. I just raised it as a potential issue based on what I remember reading on discourse / issues. Probably not worth worrying about.

Reply:
I've checked the seeds on my computer on obtained these results.
julia> @sync @parallel for i in 1:5
       println(Base.Random.GLOBAL_RNG.seed)
       end
	From worker 2:	UInt32[0x8dca2c1e, 0xf8d0c0c2, 0x41bd9aef, 0x8f93d3b6]
	From worker 2:	UInt32[0x8dca2c1e, 0xf8d0c0c2, 0x41bd9aef, 0x8f93d3b6]
	From worker 3:	UInt32[0xdd8a8792, 0x85d04ba8, 0xc03a6442, 0x7ee40770]
	From worker 5:	UInt32[0x9898716c, 0x6206fe0f, 0x4fc67721, 0xfc8ace37]
	From worker 4:	UInt32[0xdf5c2a4b, 0xae7c907a, 0x275310b2, 0xbc04a35b]

It think its safe to confirm that different processes have different seeds.